### PR TITLE
ENH close migrations after a time

### PR DESCRIPTION
**Pull request by beckermr**
I am starting from @CJ-Wright's branch and then adding the time thing.

### CODE REVIEW

File: conda_forge_tick/migrators/core.py [hunk 1]
```diff
@@ -443,6 +445,51 @@ def __init__(
...
+            try:
+                payload = self.graph.nodes[node]["payload"]
+            except KeyError as e:
+                print(node)
+                raise e
+
+            # we don't need to look at archived feedstocks
+            # they are always "migrated"
+            if payload.get("archived", False):
+                continue
```
Comment (CJ-Wright) [inline conda_forge_tick/migrators/core.py:460]: Is it worthwhile pulling this out as a generator that we walk through, to reduce code copy paste?

File: conda_forge_tick/migrators/core.py [hunk 2]
```diff
@@ -443,6 +445,51 @@ def __init__(
...
+                ts = (
+                    payload.get("PRed", [])[pr_index]
+                    .get("PR", {})
+                    .get("updated_at", None)
+                )
+                state = payload.get("PR", {"state": "open"}).get("state", "")
+                if ts is not None and state == "open":
+                    now = datetime.datetime.now(datetime.timezone.utc)
+                    ts = dateutil.parser.parse(ts)
+                    if now - ts < datetime.timedelta(days=30):
```
Comment (CJ-Wright) [inline conda_forge_tick/migrators/core.py:479]: Should we get the timedelta from the migration yaml itself? (And default to some default value?)

Comment (beckermr) [reply conda_forge_tick/migrators/core.py:479]: The time delta is for the PRs from the migration, not when the migrator was started. We want to close when all PRs have been issued and every PR is at least 30 days old.

File: conda_forge_tick/migrators/core.py (hunk 1 above)
Comment (beckermr) [reply conda_forge_tick/migrators/core.py:460]: yup we can do that!

**Comment by beckermr** [comment]
ok @CJ-Wright ready for another look!

File: conda_forge_tick/migrators/core.py (hunk 2 above)
Comment (CJ-Wright) [reply conda_forge_tick/migrators/core.py:487]: Sorry. I meant more should this be a configurable field in the migration yaml. For instance some migrations may want to be longer/shorter on that time. It isn't a desperately needed feature though so we can punt if you'd like.

Comment (beckermr) [reply conda_forge_tick/migrators/core.py:487]: Let's punt I think. 30 days is a fail-safe really and this only applies to migrations which have issued all PRs.

**Review by CJ-Wright — APPROVED** [review]
LGTM thank you!

**Pull request merged**